varname.py

```diff
@@ -1,57 +1,60 @@
 """Dark magics about variable name in python"""
+import os
+import sys
 import ast
 import dis
-import sys
 import inspect
 import warnings
 from typing import Callable, List, Union, Tuple, Any, Optional
-from types import FrameType, CodeType, ModuleType
-from collections import namedtuple as standard_namedtuple
-from functools import wraps, lru_cache
+from types import FrameType, CodeType, FunctionType, ModuleType
+from functools import wraps, lru_cache, partial
 
 import executing
 
-__version__ = "0.5.6"
+__version__ = "0.6.0"
 __all__ = [
-    "VarnameRetrievingError", "varname", "will", "inject_varname",
-    "register", "inject", "nameof", "namedtuple", "Wrapper", "debug"
+    "VarnameRetrievingError", "varname", "will",
+    "register", "nameof", "Wrapper", "debug"
 ]
 
+IgnoreType = List[Union[ModuleType, FunctionType, Tuple[ModuleType, str]]]
+
 # To show how the desired frame is selected and other frames are skipped
 DEBUG = False
 
 class VarnameRetrievingError(Exception):
     """When failed to retrieve the varname"""
 
 def varname(
-        caller: int = 1,
-        ignore: Optional[
-            List[Union[ModuleType, Tuple[ModuleType, str]]]
-        ] = None,
+        frame: int = 1,
+        ignore: Optional[IgnoreType] = None,
         multi_vars: bool = False,
         raise_exc: bool = True
 ) -> Optional[Union[str, Tuple[Union[str, tuple]]]]:
     """Get the variable name that assigned by function/class calls
 
-    To debug and specify the right caller and ignore arguments, you can set
+    To debug and specify the right frame and ignore arguments, you can set
     debug on and see how the frames are ignored or selected:
 
     >>> import varname
     >>> varname.DEBUG = True
 
     Args:
-        caller: The call frame index, indicating where this function
+        frame: The call frame index, indicating where this function
             is called relative to where the variable is finally retrieved
             Frames are counted with the ignored ones being excluded.
             See `ignore` for frames to be ignored.
-        ignore: A list of modules or tuples of module and qualname that
-            you want to ignore for the intermediate calls.
-            For example, `['module']` ignores all intermediate calls
-            from `module` and its submodules, but `[(module, 'func')]`
-            only ignores the calls (qual)named `func` from `module`.
+        ignore: A list of calls to be ignored in counting the frame.
+            You can specify:
+            - A function. Note about the decorated function. You may also need
+                to ignore the original function. If the original function is
+                not avaiable anymore. You can have to specifiy it in
+                `(module, qualname)` way
+            - A module. All calls from it and its submodules will be ignored.
+            - A tuple of module and qualname. Note that the qualname in the
```

Comment:
People may be confused by the word 'qualname'. Also clarify that submodule filtering doesn't happen here.